src/sentry/apidocs/preprocessor.py

```diff
@@ -0,0 +1,17 @@
+PUBLIC_ENDPOINTS = {}
+
+# need to import extensions here so preprocessor knows they exist when ran
+from sentry.apidocs.extensions import *  # NOQA
+
+
+def custom_preprocessing_hook(endpoints):  # TODO: organize method, rename
+    filtered = []
+    for (path, path_regex, method, callback) in endpoints:
+
+        if (
+            callback.__name__ in PUBLIC_ENDPOINTS
+            and method in PUBLIC_ENDPOINTS[callback.__name__]["methods"]
+        ):
+            filtered.append((path, path_regex, method, callback))
+
+    return filtered
```

Comment:
We could factor the `filtered` variable out by making this a `yield` generator, i.e.

```python
def custom_preprocessing_hook(endpoints):
    for (path, path_regex, method, callback) in endpoints:
        if (...):
            yield (path, path_regex, method, callback)
```

(I peeked ahead, and the Spectacular code that consumes this should be fine with a lazy iterable.)